test/unit/linear_algebra/test_tridiagonal_solver.cpp

```diff
@@ -16,10 +16,10 @@ typedef std::vector<double> vecd;
 typedef TridiagonalMatrix<float> trid_matf;
 typedef std::vector<float> vecf;
 
-const std::size_t size = 100000000; // size of the system to test
-const double tol_dp = 1e-14;        // tolorance for double
-const float tol_sp = 1e-5; // tolorance for single point floating precision
-const int norm_order = 1;  // L2 norm for computing error
+const std::size_t size = 10; // size of the system to test
+const double tol_dp = std::numeric_limits<double>::epsilon();
+const float tol_sp = std::numeric_limits<float>::epsilon();
+const int norm_order = 2; // L2 norm for computing error
```

Comment:
if you change to / stay with the relative error, probably the `norm_order` option is no longer needed?